### PR TITLE
Feature/tk18 develop alert when email is not valid for account recovery

### PR DESCRIPTION
**Pull request by MarinaFX**
# What was done? ✅
 - Step 1: Implemented recovery account error alert cases
 - Step 2: project refactoring (folder + file name adjustments and improvementes)
 - Step 3: localized alerts

# Description 📝
## Project Info 📈
 - US **15**
 - Task 
   - ID: **17/18/90/91/92/93/94**
   - Title: **Develop Alert when account does not exist for account recovery**
   - Title: **Develop Alert when email is not valid for account recovery**
   - Title: **Develop Alert for unmatched passwords for account recovery**
   - Title: **Develop alert for Successful password reset for account recovery**
   - Title: **Develop alert for successful email sent to user**
   - Title: **Develop alert for empty fields for account recovery (both screens)**
   - Title: **Develop alert for Internal Server error for account recovery**
  
# Evidence 🕵️‍♀️
## **Screenshots/Videos/Figma** 📱
<video src="https://user-images.githubusercontent.com/38846018/195147105-f4071ee6-a8d5-4ba9-8350-ec7724673630.mov" width=414 controls>

### CODE REVIEW

File: Fiero.xcodeproj/xcshareddata/xcschemes/Fiero.xcscheme [hunk 1]
```diff
@@ -54,7 +54,7 @@
       buildConfiguration = "Debug"
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
-      language = "pt-BR"
+      language = "en"
```
Comment (illuminatyy) [inline Fiero.xcodeproj/xcshareddata/xcschemes/Fiero.xcscheme:57]: Porque foi alterado pra inglês?

File: Fiero/ViewModel/UserViewModel.swift [hunk 2]
```diff
@@ -306,12 +312,11 @@ class UserViewModel: ObservableObject {
...
         }
         """
-        
+        print(json)
         let request = makePATCHRequest(json: json, scheme: "http", port: 3333, baseURL: FieroAPIEnum.BASE_URL.description, endPoint: UserEndpointEnum.RESET_PASSWORD.description, authToken: "")
         
-        
         let operation = self.client.perform(for: request)
-            .tryMap({ $1 as? HTTPURLResponse })
+            .tryMap({ $1 as! HTTPURLResponse })
```
Comment (illuminatyy) [inline Fiero/ViewModel/UserViewModel.swift:319]: É mais uma dúvida mesmo, mas porque foi tirado o opcional? Não é melhor do que forçar?

File: Fiero/ViewModel/UserViewModel.swift [hunk 3]
```diff
@@ -325,23 +330,30 @@ class UserViewModel: ObservableObject {
                         print("Successfully created request to reset password endpoint")
                 }
             }, receiveValue: { [weak self] response in
-                guard let response = response else {
-                    print("There was an error while trying to reset the password: \(String(describing: response?.statusCode))")
-                    return
-                }
+//                guard let response = response else {
```
Comment (illuminatyy) [inline Fiero/ViewModel/UserViewModel.swift:333]: É válido deixar esse código comentado?

File: Fiero.xcodeproj/xcshareddata/xcschemes/Fiero.xcscheme (hunk 1 above)
Comment (MarinaFX) [reply Fiero.xcodeproj/xcshareddata/xcschemes/Fiero.xcscheme:57]: Ja tava em ingles antes, provavelmente eu comitei quando alterei pra portugues e ficou assim. Mas não eh algo que influencie em rodar o projeto

File: Fiero/ViewModel/UserViewModel.swift (hunk 2 above)
Comment (MarinaFX) [reply Fiero/ViewModel/UserViewModel.swift:319]: Apenas pra nao fazer um guard let no sink da chamada. Nesse caso eu cheguei a botar um force pq é igual ao dequeueReusableCell no UIKit, queremos forçar aqui pq provavelmente vai ter um erro no servidor se nao retornar um HTTPUrlResponse

File: Fiero/ViewModel/UserViewModel.swift (hunk 3 above)
Comment (MarinaFX) [reply Fiero/ViewModel/UserViewModel.swift:333]: nao, bem lembrado.

File: en.lproj/Localizable.strings [hunk 4]
```diff
@@ -223,16 +232,28 @@ We can't get your challenges now, try again";
 "accountNotFoundMessage" = "We didn't find any account connected to this email, would you like to Signup?";
 "accountNotFoundPrimaryButton" = "Create account";
 
+//MARK: Screens
+//MARK: Insert code and password
+"verificationcationCodeTitleLabel" = "Insert the code we've just sent to your email and your new password";
+"verificationCodeTextFieldPlaceholder" = "Validation Code";
+"passwordTextFieldPlaceholder" = "New Password";
+"confirmPasswordTextFieldPlaceholder" = "Confirm new password";
+"changePasswordButtonLabel" = "Change password";
```
Comment (LucasJusto) [inline en.lproj/Localizable.strings:241]: Why in "New Password" and "Verification Code" all words start uppercase but in "Change password" and "Confirm new password" it doesnt?

File: pt-BR.lproj/Localizable.strings [hunk 5]
```diff
@@ -229,16 +238,28 @@
...
 
+//MARK: Screens
+//MARK: Insert code and password
+"verificationcationCodeTitleLabel" = "Digite o código que enviamos para o seu e-mail e a nova senha!";
+"verificationCodeTextFieldPlaceholder" = "Código de validação";
+"passwordTextFieldPlaceholder" = "Nova senha";
+"confirmPasswordTextFieldPlaceholder" = "Confirme a nova senha";
+"changePasswordButtonLabel" = "Mudar senha";
+
+//MARK: Alerts
```
Comment (LucasJusto) [inline pt-BR.lproj/Localizable.strings:249]: Keep the same pattern as english about starting or not all words from the sentence with uppercase.

File: Fiero/Extensions/Enums/RecoveryAccountErrorCases.swift [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+//  Fiero
+//
+//  Created by Marina De Pazzi on 10/10/22.
+//
+
+import Foundation
+import SwiftUI
+
+//MARK: - RecoveryAccountErrorCases
+enum RecoveryAccountErrorCases {
```
Comment (LucasJusto) [inline Fiero/Extensions/Enums/RecoveryAccountErrorCases.swift:12]: At some moment you decided that all enums should have the word Enum at their declaration, but looks like you forgot it at this Enum. In my opinion it shouldn't have the word Enum in any of them. But you may like to change it here if you want the word Enum at all of them.

**Review by illuminatyy — APPROVED** [review]

**Pull request merged**